pyproject.toml

```diff
@@ -1,47 +1,89 @@
+[project]
+name = "dune-sync"
+dynamic = ['version', 'dependencies']
+
+[tool.poetry]
+name = "dune-sync"
+version = "0.2.0"
+description = ""
+authors = [
+    "Benjamin Smith <bh2smith@users.noreply.github.com>",
+    "mooster531 <mooster531@users.noreply.github.com>",
+    "Felix Leupold <masked@example.com>"
+]
+
+[tool.poetry.dependencies]
+python = "^3.13"
+dune-client = ">=1.7.7"
+pandas = "*"
+sqlalchemy = "*"
+python-dotenv = "*"
+psycopg2-binary = "*"
+tomli = "*"
```

Comment:
I think we are missing pyyaml here:

https://github.com/bh2smith/dune-sync/blob/main/requirements/prod.txt


Shouldn't we also delete our requirements directory as well?

Reply:
The contents of the `requirements` directory are still being used by the Dockerfile. I will adapt that to use poetry as well so we can remove it.

Reply:
Done in [817b0a2](https://github.com/bh2smith/dune-sync/pull/88/commits/817b0a28269dd238ae207ae1470fe2aa22fb0859)